src/generator-extension-methods.cs

```diff
@@ -0,0 +1,245 @@
+			if (generator.AttributeManager.HasAttribute<ModelAttribute> (parentType)) {
+				foreach (PropertyInfo pinfo in parentType.GetProperties (flags)) {
+					bool toadd = true;
+					var modelea = generator.GetExportAttribute (pinfo, out _);
+
+					if (modelea == null)
+						continue;
+
+					foreach (PropertyInfo exists in properties) {
+						var origea = generator.GetExportAttribute (exists, out _);
+						if (origea.Selector == modelea.Selector)
+							toadd = false;
+					}
+
+					if (toadd)
+						properties.Add (pinfo);
+				}
+			}
+		}
+
+		return properties;
+	}
+
+	public static List<MethodInfo> GatherMethods (this Type type, Generator generator)
+	{
+		return type.GatherMethods (BindingFlags.Public | BindingFlags.Instance | BindingFlags.Static, generator);
+	}
+
+	public static bool IsInternal (this MemberInfo mi, Generator generator)
+	{
+		return generator.AttributeManager.HasAttribute<InternalAttribute> (mi)
+			|| (generator.AttributeManager.HasAttribute<UnifiedInternalAttribute> (mi));
+	}
+
+	public static bool IsUnifiedInternal (this MemberInfo mi, Generator generator)
+	{
+		return (generator.AttributeManager.HasAttribute<UnifiedInternalAttribute> (mi));
+	}
+
+	public static bool IsInternal (this PropertyInfo pi, Generator generator)
+	{
+		return generator.AttributeManager.HasAttribute<InternalAttribute> (pi)
+			|| (generator.AttributeManager.HasAttribute<UnifiedInternalAttribute> (pi));
+	}
+
+	public static bool IsInternal (this Type type, Generator generator)
+	{
+		return generator.AttributeManager.HasAttribute<InternalAttribute> (type)
+			|| (generator.AttributeManager.HasAttribute<UnifiedInternalAttribute> (type));
+	}
+
+	public static List<MethodInfo> GatherMethods (this Type type, BindingFlags flags, Generator generator)
+	{
+		List<MethodInfo> methods = new List<MethodInfo> (type.GetMethods (flags));
+
+		if (generator.IsPublicMode)
+			return methods;
+
+		Type parentType = GetBaseType (type, generator);
+
+		if (parentType != generator.TypeManager.NSObject) {
+			if (generator.AttributeManager.HasAttribute<ModelAttribute> (parentType))
+				foreach (MethodInfo minfo in parentType.GetMethods ())
+					if (generator.AttributeManager.HasAttribute<ExportAttribute> (minfo))
+						methods.Add (minfo);
+		}
+
+		return methods;
+	}
+}
+
+// Fixes bug 27430 - btouch doesn't escape identifiers with the same name as C# keywords
+public static class StringExtensions {
+	public static string? GetSafeParamName (this string? paramName)
+	{
+		if (paramName is null)
+			return paramName;
+
+		if (!IsValidIdentifier (paramName, out var hasIllegalChars)) {
+			return hasIllegalChars ? null : "@" + paramName;
+		}
+		return paramName;
+	}
+
+	// Since we're building against the iOS assemblies and there's no code generation there,
+	// I'm bringing the implementation from:
+	// mono/mcs/class//System/Microsoft.CSharp/CSharpCodeGenerator.cs
+	static bool IsValidIdentifier (string? identifier, out bool hasIllegalChars)
```

Comment:
I like the out keyword implementation- a cleaner way to keep track of what is going on!